File: BackEnd/DotNet/WebDigital/WebDigital/dto/ProductDetailDto.cs
namespace WebDigital.dto
{
    public class ProductDetailDto
    {
        public int ProductId { get; set; }

        public string ProductName { get; set; } = null!;

        public string ProductIsbn { get; set; } = null!;

        public string? ProductDescriptionShort { get; set; }

        public string? ProductDescriptionLong { get; set; }

        public string? ProductImage { get; set; }

        public decimal ProductBaseprice { get; set; }

        public decimal? ProductOfferprice { get; set; }

        public DateOnly? ProductOffPriceExpirydate { get; set; }

        public decimal? DiscountPercent { get; set; }

        public decimal? RentPerDay { get; set; }

        public int? MinRentDays { get; set; }

        public bool IsRentable { get; set; }

        public bool IsLibrary { get; set; }

        public string? AuthorName { get; set; }

        public string? GenreName { get; set; }

        public string? LanguageName { get; set; }

        public string? PublisherName { get; set; }

        public string? TypeName { get; set; }
    }
}

File: BackEnd/Java/product-service/src/main/java/com/digitalshelf/productservice/dto/ProductDetailDto.java
package com.digitalshelf.productservice.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductDetailDto {
    private Integer productId;
    private String productName;
    private String productIsbn;
    private String productDescriptionShort;
    private String productDescriptionLong;
    private String productImage;
    private BigDecimal productBaseprice;
    private BigDecimal productOfferprice;
    private LocalDate productOffPriceExpirydate;
    private BigDecimal discountPercent;
    private BigDecimal rentPerDay;
    private Integer minRentDays;
    private boolean isRentable;
    private boolean isLibrary;

    private String authorName;
    private String genreName;
    private String languageName;
    private String publisherName;
    private String typeName;

    private Integer authorId;
    private Integer genreId;
    private Integer languageId;
    private Integer publisherId;
    private Integer typeId;
}